src/lib.rs

```diff
@@ -269,3 +293,128 @@ impl<'a> AsRef<[u8]> for Diff<'a> {
         self.bytes.as_ref()
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use std::{sync::Arc, time::Duration};
+
+    use iroh::protocol::Router;
+    use loro::{EventTriggerKind, LoroDoc};
+    use tracing::{error_span, info, Instrument};
+    use tracing_test::traced_test;
+
+    use crate::{IrohLoroProtocol, SyncMode};
+
+    async fn setup_node(doc: LoroDoc) -> anyhow::Result<(Router, IrohLoroProtocol)> {
```

Comment:
Instead of `anyhow`, mind using `TestResult` from the `testresult` crate?
It basically just panics when there's any `Err`, but you keep the `?` syntax.
Gives you the best of both worlds: The stacktrace points to the line where the error happens (in the test), and you get to keep the nice syntax sugar. We've started using it in other places in iroh, too.